### PR TITLE
Add section at top of page to show what You are playing!

### PR DESCRIPTION
**Pull request by ghostlandr**
Also refactored out some of the lastfm specific functions into lastfm.js
to make re-use between components a bit simpler. If it ended up more
complicated let me know.

@esalter-va

### CODE REVIEW

File: static/js/app.js [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
         setInterval(
-            (function(scope) {
+            function(scope) {
                 return function() {
                     for (friend of scope.friends) {
-                        scope.getRecentTracks(friend.name);
+                        getRecentTracks(friend.name, scope.recentTracksCallback, scope);
                     }
                 }
-            })(this), 10000
```
Comment (ghostlandr) [inline static/js/app.js]: Ohh I see how this was working before! I'll change it back.

**Comment by evansalter** [comment]
Another way this could be done, that might end up with less new code:

1. When page is loaded, get the current user's info
1. Right before the `for (friend of scope.friends)` loop, append the current user's information to `this.friends`
1. The current code will then loop through the array of friends (which includes yourself), and get the recent tracks
1. Modify the sorting function to always put the current user on top.

Not sure if it's any better, just might be something to think about.

**Comment by ghostlandr** [comment]
That way seems a lot less simple to me. I thought about trying to just munge "you" into all the spots we get data but in the end it seemed to make more sense to make a separate component that for now is just kind of a copy paste but some day could be something better, and it doesn't depend on the current friend list code for anything.

**Comment by evansalter** [comment]
Fair enough.  I might squeeze in a new component for displaying a friend and their recent tracks, then we can call the same thing from `friend-list` and `you`

**Comment by ghostlandr** [comment]
That'd be sweet! You're suggesting I do it? Or you want to tackle 'er? Probably makes sense that these can share a lot or we can just make a "last fm user" type of component that everyone uses.

**Comment by ghostlandr** [comment]
Well, that's done

**Comment by evansalter** [comment]
Dat live-updating.  Thanks!

**Comment by evansalter** [comment]
I guess we could have just re-used `friend-list` with a list of one friend, but whatever.  Doesn't really make a difference

**Review by evansalter — APPROVED** [review]

**Comment by ghostlandr** [comment]
I think in the end this is a net-positive.

**Comment by evansalter** [comment]
[Add You section to Scrobblin with Friends](https://trello.com/c/JYlXPXXp/337-scrobblin-with-friends-quick-i-swear)

**Pull request merged**